File: raspberry/SmartHome/app/src/main/java/raspberry/smarthome/database/store/RPInfoStorage.java
package raspberry.smarthome.database.store;

import com.google.firebase.database.DatabaseReference;

import raspberry.smarthome.database.store.listeners.RPInfoListener;

public interface RPInfoStorage {

    void postRaspberryIp(String ip);

    void postRaspberryIp(String ip, DatabaseReference.CompletionListener listener);

    void getRaspberryInfo(RPInfoListener listener);

}

File: smarthomelibary/DataLibrary/datalibrary/src/main/java/smarthome/datalibrary/database/store/RPInfoStorage.kt
package smarthome.datalibrary.database.store

import com.google.android.gms.tasks.OnFailureListener
import com.google.android.gms.tasks.OnSuccessListener
import smarthome.datalibrary.database.constants.Constants.defFailureListener
import smarthome.datalibrary.database.constants.Constants.defSuccessListener
import smarthome.datalibrary.database.store.listeners.RPInfoListener

interface RPInfoStorage {
    fun postRaspberryIp(ip: String, successListener: OnSuccessListener<Void> = defSuccessListener, failureListener: OnFailureListener = defFailureListener)
    fun postRaspberryPort(port: String, successListener: OnSuccessListener<Void> = defSuccessListener, failureListener: OnFailureListener = defFailureListener)
    fun getRaspberryInfo(listener: RPInfoListener)
}

